src/quo2/components/buttons/button/style.cljs

```diff
@@ -53,54 +49,67 @@
                       56 12
                       40 12
                       32 10
-                      8))})
+                      24 8
+                      12))})
 
 (defn style-container
-  [{:keys [size disabled border-radius background-color border-color icon-only? icon-above width
+  [{:keys [size disabled? border-radius background-color border-color icon-only? icon-top
            icon-left icon-right]}]
   (merge {:height             size
           :align-items        :center
           :justify-content    :center
-          :flex-direction     (if icon-above :column :row)
+          :flex-direction     (if icon-top :column :row)
           :padding-horizontal (when-not (or icon-only? icon-left icon-right)
                                 (case size
-                                  56 16
+                                  56 (if border-color 10 11)
                                   40 16
                                   32 12
-                                  8))
+                                  24 7
+                                  16))
           :padding-left       (when-not (or icon-only? icon-left)
                                 (case size
-                                  56 16
+                                  56 nil
```

Comment:
Great point @J-Son89. If there's no default value, the ideal scenario to me is to log and return nil. Unmatched conditions that will look wrong shouldn't go unnoticed. Of course, one may argue that nobody will see the logs, but it surely helps devs to more proactively catch these issues instead of real users, which would be bad.

Trusting that the code will pass only valid sizes now and in the future is a recipe for user bugs with `case`, so I would avoid that.

This idea to return nil to avoid the exception with `case` and log something if the unmatched case is unexpected is a good practice in general, so I think it doesn't hurt to do it. To be fair, I haven't done this in status-mobile, but I'll keep this practice in mind.

Reply:
Oh well this is quite specific to setting up this component because here padding horizontal can clash with padding left and padding right. Ideally this code could be written to just avoid the use of padding horizontal here but that would involve me redesigning the button styling and so I don't want to invest that effort as there is a lot variations to check and I have invested a lot of time in this component already. But there is definitely a cleaner approach here